integration-tests/cypress/commands/api/device.ts

```diff
@@ -0,0 +1,539 @@
+import { getTestName } from "../names";
+import {
+  v1ApiPath,
+  saveCreds,
+  getCreds,
+  uploadFileRequest,
+  checkRequestFails,
+  makeAuthorizedRequest,
+  makeAuthorizedRequestWithStatus,
+  sortArrayOn,
+  sortArrayOnTwoKeys
+} from "../server";
+import { logTestDescription } from "../descriptions";
+const url = require("url");
+
+Cypress.Commands.add(
+  "recordEvent",
+  (
+    camera: string,
+    type: string,
+    details = {},
+    date = new Date(),
+    log = true
+  ) => {
+    const data = {
+      dateTimes: [date.toISOString()],
+      description: { type: type, details: details }
+    };
+    logTestDescription(
+      `Create ${type} event for ${camera} at ${date}`,
+      { data: data },
+      log
+    );
+    makeAuthorizedRequest(
+      {
+        method: "POST",
+        url: v1ApiPath("events"),
+        body: data
+      },
+      camera
+    );
+  }
+);
+
+Cypress.Commands.add(
+  "apiCreateCamera",
+  (cameraName: string, group: string, saltId: number = null, log = true, statusCode: number = 200) => {
+    logTestDescription(
+      `Create camera '${cameraName}' in group '${group}' with saltId '${saltId}'`,
+      {
+        camera: cameraName,
+        group: group,
+	saltId: saltId
+      },
+      log
+    );
+
+    const request = createCameraDetails(cameraName, group, null, saltId);
+    if(statusCode==200) {
+      cy.request(request).then((response) => {
+        const id = response.body.id;
+        saveCreds(response, cameraName, id);
+      });
+    } else {
+      checkRequestFails(request);
+    };
+  }
+);
+
+Cypress.Commands.add(
+  "apiDeviceReregister",
+  (oldName: string, newName: string, newGroup: string, password: string = null, keepName:boolean = false, statusCode: number = 200) => {
+    var uniqueName;
+    logTestDescription(
+      `Reregister camera '${newName}' in group '${newGroup}'`,
+      {
+        camera: newName,
+        group: newGroup
+      },
+      true
+    );
+
+    if(password==null) {
+      password="p"+getTestName(newName);
+    };
+
+    if(keepName==false) {
+      uniqueName=getTestName(newName);
+    } else {
+      uniqueName=newName;
+    };
+
+    const data = {
+      newName: uniqueName,
+      newPassword: password,
+      newGroup: getTestName(newGroup)
+    };
+    const fullUrl = v1ApiPath('devices/reregister');
+
+    makeAuthorizedRequestWithStatus(
+      {
+        method: "POST",
+        url: fullUrl,
+        body: data
+      },
+      oldName,
+      statusCode
+    ).then((response) => {
+      const id = response.body.id;
+      saveCreds(response, newName, id);
+    });
+  }
+);
+
+
+Cypress.Commands.add(
```

Comment:
I think this was already there, but it seems to be strange that it's a single command feels like it should be tested in a test

Reply:
yes - pre-existing and used by Clare's code. Will rework to match the standard test structure.

Reply:
Removed this duplicate function and refactored tests to use the standard createDevice function and check for correct fail result